pkg/diagnostics/service_metrics.go

```diff
@@ -0,0 +1,144 @@
+package diagnostics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+// ServiceMetrics holds dapr runtime metric monitoring methods
+type ServiceMetrics struct {
+	componentLoaded               *stats.Int64Measure
+	componentInitCompleted        *stats.Int64Measure
+	componentInitFailed           *stats.Int64Measure
+	mtlsInitCompleted             *stats.Int64Measure
+	mtlsInitFailed                *stats.Int64Measure
+	mtlsWorkloadCertRotated       *stats.Int64Measure
+	mtlsWorkloadCertRotatedFailed *stats.Int64Measure
+
+	// Metrics Tags
+	appIDTag      tag.Key
+	componentTag  tag.Key
+	failReasonTag tag.Key
+
+	ctx context.Context
+}
+
+func newView(measure stats.Measure, keys []tag.Key, aggregation *view.Aggregation) *view.View {
+	return &view.View{
+		Name:        measure.Name(),
+		Description: measure.Description(),
+		Measure:     measure,
+		TagKeys:     keys,
+		Aggregation: aggregation,
+	}
+}
+
+// NewServiceMetrics returns ServiceMetrics instance with default service metric6 stats
+func NewServiceMetrics() *ServiceMetrics {
+	appIDTag, _ := tag.NewKey("app_id")
+	componentTag, _ := tag.NewKey("component")
+	failReasonTag, _ := tag.NewKey("reason")
+
+	return &ServiceMetrics{
+		// Runtime Component metrics
+		componentLoaded: stats.Int64(
+			"runtime/component/loaded",
+			"The number of successfully loaded components",
+			stats.UnitDimensionless),
+		componentInitCompleted: stats.Int64(
+			"runtime/component/init_total",
+			"The number of initialized components",
+			stats.UnitDimensionless),
+		componentInitFailed: stats.Int64(
+			"runtime/component/init_fail_total",
+			"The number of component initialization failures",
+			stats.UnitDimensionless),
+
+		mtlsInitCompleted: stats.Int64(
+			"runtime/mtls/init_total",
+			"The number of successful mTLS authenticator initialization.",
+			stats.UnitDimensionless),
+		mtlsInitFailed: stats.Int64(
+			"runtime/mtls/init_fail_total",
+			"The number of mTLS authenticator init failures",
+			stats.UnitDimensionless),
+		mtlsWorkloadCertRotated: stats.Int64(
+			"runtime/mtls/workload_cert_rotated_total",
+			"The number of mTLS authenticator init failures",
```

Comment:
This is a dupe of line `66`

Reply:
this is WIP :) I will update it later. I just want to check metrics are emitting now. I will ask you to review this pr when I complete it.

Reply:
Ok :)